cypress/integration/spec.js

```diff
@@ -76,31 +88,90 @@ describe('Chart Explorer', function() {
     ).click()
     cy.get('[data-test=rm-encoding]').click()
 
-    // check for editor existence
-    cy.get('#configure-tabs-tab-editor').click()
-    cy.get('.monaco-editor')
-    cy.get('#configure-tabs-tab-builder').click()
-
     // check that titles work
     cy.get('[data-test=chart-title]').type('test title')
     cy.get('svg .role-title').contains('test title')
 
     // check that chart exists
     cy.get('[data-test=vega-embed]')
-
     cy.screenshot('page with viz')
 
-    // check download options
+    cy.get('[data-test=license-open]').click()
+    cy.get('[data-test=license-text]').contains('cypress license text')
+  })
+
+  it('Should be able to edit chart using editor', function() {
+    cy.get('#configure-tabs-tab-editor').click()
+
+    cy.get('.inputarea')
+      .type('{ctrl}f')
+      .focused()
+      .type('"title": "test 123",')
+
+    cy.get('#configure-tabs-tab-builder').click()
+
+    cy.get('svg .role-title').contains('test 123')
+  })
+
+  it('Should be able to download a chart', function() {
     cy.get('#dropdown-download').click()
 
+    // download as a vega-lite file format
     cy.get('.open > .dropdown-menu > :nth-child(3) > a').trigger('mousedown')
     cy.get('.open > .dropdown-menu > :nth-child(3) > a').should($m => {
       expect($m).to.have.length(1)
       expect($m.attr('href')).to.match(/^blob/)
     })
-    cy.get('#dropdown-download').click()
 
-    // check download as insight
+    // download as a vega file format
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').trigger('mousedown')
+    cy.get('.open > .dropdown-menu > :nth-child(2) > a').should($m => {
```

Comment:
Why `.trigger('mousedown')` instead of `.click()` like in the later `it` block?

Reply:
I had thought initially that the creator of these tests was just trying out other ways of performing the same thing. But after converting it to a click event I realized that they are actually different. 

Having it as `.trigger('mousedown')` will perform the action just enough to assert that it does as it's intended to do without actually downloading the file. When I change it to `click` though, it actually performs the download. The thing is I don't actually need to perform the download to assert that it works as expected so I think it makes sense to just leave it the way it is for the sake of performance. 

Thoughts?

Reply:
very cool!! yes, then i think leaving it as is is definitely the preferred route 🙌

Reply:
>Having it as `.trigger('mousedown')` will perform the action just enough to assert that it does as it's intended to do without actually downloading the file. When I change it to `click` though, it actually performs the download. The thing is I don't actually need to perform the download to assert that it works as expected so I think it makes sense to just leave it the way it is for the sake of performance.


How can we capture this insight into something that everyone can learn, without having to learn it the hard way?

Reply:
Ok after looking into it more I realized this is specific only to this case. We are passing a onMouseDown handler which is why just doing onMouseDown actually does something as opposed to having to do a click which performs both a mousedown and mouseup event.

```  
handleMouseDown = async (e: SyntheticMouseEvent<HTMLElement>) => {
    const { currentTarget } = e
    const { getDownloadUrl, baseName, extension } = this.props


    const url = await getDownloadUrl()
    currentTarget.setAttribute('href', url)
    currentTarget.setAttribute('download', getDownloadName(baseName, extension))
  }

```